pacta/cj_build_tag.sh

```diff
@@ -0,0 +1,45 @@
+clones="PACTA_analysis create_interactive_report StressTestingModelDev pacta-data"
+url="git@github.com:2DegreesInvesting/"
+
+for repo in ${clones}
+do
+    remote="${url}${repo}.git"
+    git clone -b master ${remote} --depth 1
+    echo "--"
+done
+
+for repo in ${clones}
+do
+    echo "${repo} HEAD sha:"
+    git -C "${repo}" rev-parse HEAD
+    echo "--"
+done
+
+docker rmi 2dii_pacta
```

Comment:
This may fail if there is more than one tag for the 2dii_pacta image:

```bash
➜  pacta git:(pacta-web-build) ✗ docker images                                
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
2dii_pacta          0.0.0.9000          43c79a68e414        3 minutes ago       3.36GB
2dii_pacta          1                   43c79a68e414        3 minutes ago       3.36GB
2dii_pacta          2                   43c79a68e414        3 minutes ago       3.36GB
maurolepore/pacta   0.0.1               358e8fe546f1        26 hours ago        3.71GB
2dii/r-packages     latest              2d1d6a7e7ade        44 hours ago        1.76GB
2dii/system         latest              9790f300c9cd        44 hours ago        1.26GB
ubuntu              18.04               56def654ec22        5 weeks ago         63.2MB

➜  pacta git:(pacta-web-build) ✗ docker rmi 2dii_pacta
Error: No such image: 2dii_pacta

➜  pacta git:(pacta-web-build) ✗ docker rmi 2dii_pacta:2
Untagged: 2dii_pacta:2

➜  pacta git:(pacta-web-build) ✗ docker rmi 2dii_pacta:1
Untagged: 2dii_pacta:1

➜  pacta git:(pacta-web-build) ✗ docker images          
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
2dii_pacta          0.0.0.9000          43c79a68e414        4 minutes ago       3.36GB
maurolepore/pacta   0.0.1               358e8fe546f1        26 hours ago        3.71GB
2dii/r-packages     latest              2d1d6a7e7ade        44 hours ago        1.76GB
2dii/system         latest              9790f300c9cd        44 hours ago        1.26GB
ubuntu              18.04               56def654ec22        5 weeks ago         63.2MB
```

Reply:
crap.... any ideas around that?

Reply:
I think it's no big deal -- a new tag overwrites the old one. We may either remove this line, leave it as is and just stare at the error message, or grep on "2dii_pacta*" and remove all images -- but that's too intrusive I think.

Reply:
Notice how I tag a new image with a preexisting tag, first from 2dii_pacta and later form maurolepore/pacta -- just to show how the image id changes depending on the source image:

```bash
➜  pacta git:(pacta-web-build) ✗ docker tag 2dii_pacta:0.0.1.9000 2dii_pacta:1          

➜  pacta git:(pacta-web-build) ✗ docker images                                
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
2dii_pacta          0.0.1.9000          90ad58717379        About an hour ago   3.36GB
2dii_pacta          1                   90ad58717379        About an hour ago   3.36GB
maurolepore/pacta   0.0.1               358e8fe546f1        28 hours ago        3.71GB
2dii/r-packages     latest              2d1d6a7e7ade        46 hours ago        1.76GB
2dii/system         latest              9790f300c9cd        46 hours ago        1.26GB
ubuntu              18.04               56def654ec22        5 weeks ago         63.2MB

➜  pacta git:(pacta-web-build) ✗ docker tag 2dii_pacta:1 2dii_pacta:0.0.1.9000 

➜  pacta git:(pacta-web-build) ✗ docker images                                
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
2dii_pacta          0.0.1.9000          90ad58717379        About an hour ago   3.36GB
2dii_pacta          1                   90ad58717379        About an hour ago   3.36GB
maurolepore/pacta   0.0.1               358e8fe546f1        28 hours ago        3.71GB
2dii/r-packages     latest              2d1d6a7e7ade        46 hours ago        1.76GB
2dii/system         latest              9790f300c9cd        46 hours ago        1.26GB
ubuntu              18.04               56def654ec22        5 weeks ago         63.2MB

➜  pacta git:(pacta-web-build) ✗ # Above, notice the number of tags 0.0.1.9000 (only one) and the image id compared to the tag 1

➜  pacta git:(pacta-web-build) ✗ docker tag maurolepore/pacta:0.0.1 2dii_pacta:0.0.1.9000                        

➜  pacta git:(pacta-web-build) ✗ docker images                                           
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
2dii_pacta          1                   90ad58717379        About an hour ago   3.36GB
2dii_pacta          0.0.1.9000          358e8fe546f1        28 hours ago        3.71GB
maurolepore/pacta   0.0.1               358e8fe546f1        28 hours ago        3.71GB
2dii/r-packages     latest              2d1d6a7e7ade        46 hours ago        1.76GB
2dii/system         latest              9790f300c9cd        46 hours ago        1.26GB
ubuntu              18.04               56def654ec22        5 weeks ago         63.2MB

➜  pacta git:(pacta-web-build) ✗ # Above, notice the number of tags 0.0.1.9000 (still one) and the image id compared to maurolepore/pacta


```

Reply:
versioning is definitely a better idea... like auto-incrementing on the highest tag/version that already exists?

what I did (`docker rmi 2dii_pacta`) was just the nuclear way of making sure I was loading/exporting the right thing

Reply:
we can get rid of this `docker rmi 2dii_pacta`... it is not really necessary, and it was meant to make things "safer", but it seems it's possible it makes things less safe/stable under certain conditions, so I think maybe just get rid of it. @maurolepore ?

Reply:
Yeah, I can't confirm right now but this line might relate to leaving a zombie image behind.

Reply:
eventually, I replaced this line with
`docker rmi --force $(docker images -q '2dii_pacta' | uniq)`
which searches for all images named "2dii_pacta" (regardless of tag), returns just the id/s of the image/s it finds, and `uniq`'s that list before sending it to `docker rmi`. This still causes a warning message if no images exist named "2dii_pacta", but the script is still able to continue as expected, so I think this is an adequate solution for now.